### PR TITLE
Add a simple Makefile.

### PR DESCRIPTION
**Pull request by bjornt**
This adds a simple makefile that makes it easier to build the charm and push it to the charm store.

When pushing the charm to the store it also adds a repo-info file so that it's easier to track which version of the charm that is pushed.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+
+.PHONY: help
+help: ## Print help about available targets
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+.PHONY: charm-build
+charm-build:  ## Build the charm
+	rm -rf $(CHARM_OUTPUT)
+	INTERFACE_PATH=interfaces charm build -s $(CHARM_SERIES) -o $(CHARM_OUTPUT)
```
Comment (albertodonato) [inline Makefile]: Why do you need INTERFACE_PATH?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+.PHONY: charm-push
+charm-push: charm-build ## Push the rendered charm to the charm store
+	echo -n "commit-sha-1: "  > $(RENDERED_CHARM_DIR)/repo-info
+	echo $(shell git rev-parse HEAD) >> $(RENDERED_CHARM_DIR)/repo-info
```
Comment (albertodonato) [inline Makefile]: This can be just `git rev-parse HEAD >> $(RENDERED_CHARM_DIR)/repo-info`

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	charm push $(RENDERED_CHARM_DIR) $(RENDERED_CHARM_URI)
+
+$(RENDERED_CHARM_DIR):
+	$(MAKE) charm-build
```
Comment (albertodonato) [inline Makefile]: can't you just depend on charm-build here? `$(RENDERED_CHARM_DIR): charm-build`

File: Makefile (hunk 1 above)
Comment (bjornt) [inline Makefile]: No reason, just copy and paste from another charm. Removed it.

File: Makefile (hunk 2 above)
Comment (bjornt) [inline Makefile]: Indeed. I tried to to something else before, but now it can be as you suggest. Fixed.

File: Makefile (hunk 3 above)
Comment (bjornt) [inline Makefile]: Probably. This was also copy and pasted from another charm. I removed the target altogether, since  it's not used.

**Pull request merged**